theano/tensor/nnet/corr_gemm.c

```diff
@@ -172,50 +171,61 @@ PyArrayObject* corrMM(PyArrayObject* bottom,
     }
 
     // Create temporary columns
-    npy_intp col_dim[2];
-    col_dim[0] = (npy_intp)(nChannels * kW * kH);
-    col_dim[1] = (npy_intp)(topHeight * topWidth);
-    PyArrayObject* col = (PyArrayObject*)PyArray_EMPTY(2,
-		                           col_dim,
-                                           PyArray_TYPE(top),
-					   0);
+    const int max_threads = %(omp_max_threads)s < batchSize ? %(omp_max_threads)s : batchSize;
+
+    npy_intp col_dim[3];
+    col_dim[0] = (npy_intp)max_threads;
+    col_dim[1] = (npy_intp)(nChannels * kW * kH);
+    col_dim[2] = (npy_intp)(topHeight * topWidth);
+    PyArrayObject* col = (PyArrayObject*)PyArray_ZEROS(3,
```

Comment:
Hi nouiz,

When running in single thread, using PyArray_ZEROS is a little bit slow than using PyArray_EMPTY, this is what I found from the test result, but haven't figured out why these two APIs have such performance difference. I looked into their codes, the only difference between them is PyArray_ZEROS could use multi-thread (when enabled) to zero the memory.

Here's the snippet of profiling of AlexNet workload when `max_threads=1`
**Calling PyArray_ZEROS**
```
 <% time> <sum %> <apply time> <time per call> <type> <#call> <#apply> <Class name>
   45.0%    45.0%     321.487s       1.61e+00s     C      200      15   theano.tensor.nnet.corr.CorrMM
   20.0%    65.0%     142.646s       1.43e+00s     C      100       5   theano.tensor.nnet.corr.CorrMM_gradWeights
   19.9%    84.9%     141.937s       1.77e+00s     C       80       4   theano.tensor.nnet.corr.CorrMM_gradInputs
```
   
**Calling PyArray_EMPTY**
```
<% time> <sum %> <apply time> <time per call> <type> <#call> <#apply> <Class name>
   44.9%    44.9%     319.600s       1.60e+00s     C      200      15   theano.tensor.nnet.corr.CorrMM
   20.0%    64.9%     142.701s       1.43e+00s     C      100       5   theano.tensor.nnet.corr.CorrMM_gradWeights
   19.8%    84.8%     141.364s       1.77e+00s     C       80       4   theano.tensor.nnet.corr.CorrMM_gradInputs
```